master/internal/api_ntsc_intg_test.go

```diff
@@ -0,0 +1,161 @@
+//go:build integration
+// +build integration
+
+package internal
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/pkg/errors"
+
+	authz2 "github.com/determined-ai/determined/master/internal/authz"
+
+	"github.com/determined-ai/determined/master/internal/task"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/determined-ai/determined/proto/pkg/notebookv1"
+
+	"github.com/determined-ai/determined/master/internal/command"
+	"github.com/determined-ai/determined/master/internal/config"
+	"github.com/determined-ai/determined/master/internal/mocks"
+	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+)
+
+/*
+A set of tests to ensure that the NTSC APIs call the expected AuthZ methods.
+*/
+
+func setupNTSCAuthzTest(t *testing.T) (
+	*apiServer, *mocks.NSCAuthZ, model.User, context.Context,
+) {
+	api, curUser, ctx := setupAPITest(t)
+	var master *Master = api.m
+	command.RegisterAPIHandler(
+		master.system,
+		nil,
+		master.db,
+		master.rm,
+		&task.Logger{},
+	)
+	authZNSC := &mocks.NSCAuthZ{}
+	command.AuthZProvider.RegisterOverride("mock", authZNSC)
+	config.GetMasterConfig().Security.AuthZ = config.AuthZConfig{Type: "mock"}
+	return api, authZNSC, curUser, ctx
+}
+
+func setupMockNBActor(t *testing.T, master *Master) model.TaskID {
+	nbID := model.NewTaskID()
+	addr := notebooksAddr.Child(nbID)
+	ref, newCreated := master.system.ActorOf(addr, actor.ActorFunc(func(context *actor.Context) error {
+		nb := notebookv1.Notebook{Id: nbID.String()}
+		if context.ExpectingResponse() {
+			context.Respond(&apiv1.GetNotebookResponse{
+				Notebook: &nb,
+			})
+		}
+		return nil
+	}))
+	require.NotNil(t, ref)
+	require.Equal(t, newCreated, true)
+	return nbID
+}
+
+func TestTasksCountAuthZ(t *testing.T) {
+	api, authz, curUser, ctx := setupNTSCAuthzTest(t)
+	authz.On("CanGetActiveTasksCount", mock.Anything, curUser).Return(fmt.Errorf("deny"))
+	_, err := api.GetActiveTasksCount(ctx, &apiv1.GetActiveTasksCountRequest{})
+	require.Equal(t, status.Error(codes.PermissionDenied, "deny"), err)
+}
+
+func TestCanGetNTSC(t *testing.T) {
```

Comment:
Are these tests just checking errors?

Reply:
1. that we return correct errors
2. that correct rbac interfaces are being called.
3. (future pr that's open already) that we don't leak information about existence of an ntsc

Reply:
Ah ok makes sense. I was going to say something about renaming the tests, but they actually look good to me now.